launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java

```diff
@@ -363,17 +362,28 @@ public void close() throws IOException {
     /**
      * Close the connection and wait for any buffered data to be processed before returning.
```

Comment:
according to the document, shall we still call it `closeAndWait`?

Reply:
That wouldn't be accurate anymore, because the wait happens first now. `waitAndClose()` is an option but also not totally accurate. Open to suggestions.

Reply:
shall we update the document?

Reply:
Hmm, I think I may have thought of a somewhat simple way to fix the race without needing the workaround in the test. Let me try that. If that doesn't work I'll update the javadoc.

Reply:
My idea was getting too complicated for fix a rare race, so I'll just update the doc here and leave that race for another time.